### DISCUSSION TITLE
Additional checks and cleanup for NCTools

### DISCUSSION THREAD

**Discussion by ngs333** [NOAA-GFDL/FRE-NCtools#72] [Ideas]
PR #67 suggests that perhaps more of the code base needs additional sanitation. This may be for code that was modified and that there is a PR, or just code in general if time permits. PR 67s make_hgrid only core dumps in some situations, but even in situations where it doesn't core dump, Intel Inspector shows
that there are invalid memory reads. Additionally, compiling NCTools with maximum warnings shows (at least with gcc) a large number of warnings for numerous gcc apps. This issue is to discuss if there is to be any testing beyond what is commonly currently done, and/or or instructions issued concerning related testing. Here are some possible items:
a) Testing with valgrind or intel inspector and removing all critical warnings.
b) Keeping a history of apps and results that were tested under a)
c) Compiling with additional warnings. Provide instructions for doing so (Telling the user to look at the warnings; to ask for instructions. Keeping a list giving guidance of acceptable warning types. And asking users
d) Encouraging more unit tests.
e) Doing any of the above at PRs? Doing them on a schedule?

**Comment by ceblanton** [comment]
For c, could we configure the CI to reject PRs with any compiler warnings? (Once we clean up the warnings)

For a, is there any chance that the valgrind/inspector tests could be fast enough to run all the time?
